src/core/build_target.go

```diff
@@ -354,6 +356,15 @@ func (target *BuildTarget) TestDirs() string {
 	return path.Join(TmpDir, target.Label.Subrepo, target.Label.PackageName, target.Label.Name+testDirSuffix)
 }
 
+// CompleteRun completes a run and returns true if this was the last run
+func (target *BuildTarget) CompleteRun(state *BuildState) bool {
+	target.resultsMux.Lock()
+	defer target.resultsMux.Unlock()
+
+	target.completedRuns++
+	return target.completedRuns == state.NumTestRuns
```

Comment:
Is this guaranteed to always be reached eg if the worker gets killed or there is an error?

Reply:
Not necessarily, something could kill the process before it gets here. It generally should though...

Reply:
I can put this in a defer so it is less likely to fail. I think either way it would be an exceptional case where this doesn't work.